server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -485,7 +512,10 @@ private static void setPagination(SearchOptions options, SearchRequestBuilder es
       .addAggregation(termsAggregation);
 
     SearchResponse esResponse = request.get();
+    return extractAggregationTerms(aggregationKey, esResponse);
+  }
 
+  private static Set<String> extractAggregationTerms(String aggregationKey, SearchResponse esResponse) {
     Set<String> terms = new HashSet<>();
```

Comment:
Using `HashSet` looses the order used by Elasticsearch. I suggest to use `LinkedHashSet`.